root/layout/components/Search.js

```diff
@@ -46,34 +54,39 @@ const TYPE_OPTION_GROUPS = [
   },
 ];
 
-function localizedTypeOption(group, key) {
-  const option = group[key];
+function localizedTypeOption(option: SearchOptionValueT) {
   return option ? option() : '';
 }
 
+function compareTypeOptionEntries(
+  a: [string, SearchOptionValueT],
+  b: [string, SearchOptionValueT],
+) {
+  return compare(
+    localizedTypeOption(a[1]),
+    localizedTypeOption(b[1]),
+  );
+}
+
 const SearchOptions = () => (
   <select id="headerid-type" name="type">
-    {TYPE_OPTION_GROUPS.map(<TogT: {...}>(group: TogT, groupIndex) => (
-      Object.keys(group).sort(function (a, b) {
-        return compare(
-          localizedTypeOption(group, a),
-          localizedTypeOption(group, b),
-        );
-      }).map(function (key, index) {
-        const text = localizedTypeOption(group, key);
-        if (!text) {
-          return null;
-        }
-        return (
-          <option
-            key={groupIndex + '.' + index}
-            value={key}
-          >
-            {text}
-          </option>
-        );
-      })
-    ))}
+    {TYPE_OPTION_GROUPS.map((group) => (
+      Object.entries(group)
+        // $FlowIssue[incompatible-call]
```

Comment:
Can't group be typed to avoid this?

Reply:
It's already typed, but the issue is that `Object.entries` is hard-coded to return `Array<[string, mixed]>`. You can't use `mixed` without either refining it or casting it to `any` first, and both of those are worse solutions.

Reply:
Well, that sucks :(

Reply:
I suspect they'll fix the type eventually (I see they're working on improving generics checking as we speak). So the error suppression isn't so bad in the meantime: unlike `any` it doesn't prevent marking the file as strict, it only affects `incompatible-call` and not other kinds of errors, and we can run Flow with `--include-warnings` to find unneeded error suppression comments in the future.